Support: Re-enable open tickets banner

klimeryk opened the pull request:
Calypso's contact page makes a call to the Zendesk API to check if the user has open tickets, and let them know. We disabled this in https://github.com/Automattic/wp-calypso/pull/55733 to mitigate rate limiting we were experiencing with Zendesk.
This has been since fixed and we should be OK re-enabling it.

### Testing

Open the contact form. You should see a Redux action `SUPPORT_HISTORY_REQUEST` _and_ a corresponding API call matching `support-history`.

jsnajdr reviewed (approved):
This looks good if you still want to merge it and re-enable the support history requests :+1:

Found this PR when considering migrating the request to React Query.

tyxla commented:
@klimeryk if #59062 lands before this one, you'll just have to enable the `ENABLE_SUPPORT_HISTORY` contact that #59062 introduces for the same purpose.

klimeryk committed:
Support: Re-enable open tickets banner

Reverts https://github.com/Automattic/wp-calypso/pull/55733
Related https://github.com/Automattic/wp-calypso/pull/59062

klimeryk commented:
@tyxla, while testing this PR, I've run into this error:

```
Warning: Cannot update a component (`Connect(HappychatConnection)`) while rendering a different component (`InlineHelpContactViewLoaded`). To locate the bad setState() call inside `InlineHelpContactViewLoaded`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
dispatch@app:///../node_modules/redux/es/redux.js:669:28
request/<@app:///./components/data/query-ticket-support-configuration/index.jsx:16:13
section
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
div
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
InlineHelpPopover
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14 [react_devtools_backend.js:4061:25](moz-extension://3d369228-a8c1-a84d-a5f9-9fcd2848a917/build/react_devtools_backend.js)
    overrideMethod moz-extension://3d369228-a8c1-a84d-a5f9-9fcd2848a917/build/react_devtools_backend.js:4061
    React 5
    Redux 3
    batchedUpdates$1 React
    Redux 5
    batchedUpdates$1 React
    Redux 5
    batchedUpdates$1 React
    Redux 5
    batchedUpdates$1 React
    Redux 5
    batchedUpdates$1 React
    Redux 5
    batchedUpdates$1 React
    Redux 4
    e http://calypso.localhost:3000/plans/klimeryk.wordpress.com?flags=force-sympathy:2494
    dispatch app:///./state/action-log/index.js:98
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/audio/middleware.js:57
    libraryMiddleware app:///./state/lib/middleware.js:81
    analyticsMiddleware app:///./state/analytics/middleware.js:104
    enhancer Redux
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/happychat/middleware-calypso.js:221
    socketMiddleware app:///./state/happychat/middleware.js:69
    middleware app:///./state/data-layer/wpcom-api-middleware.js:73
    Redux 2
    openSocket app:///./lib/happychat/connection.js:71
    emit app:///../node_modules/socket.io-client/node_modules/component-emitter/index.js:133
    emit app:///../node_modules/socket.io-client/lib/socket.js:138
    onclose app:///../node_modules/socket.io-client/lib/socket.js:214
    exports app:///../node_modules/component-bind/index.js:21
    emit app:///../node_modules/socket.io-client/node_modules/component-emitter/index.js:133
    onclose app:///../node_modules/socket.io-client/lib/manager.js:502
    exports app:///../node_modules/component-bind/index.js:21
    emit app:///../node_modules/engine.io-client/node_modules/component-emitter/index.js:133
    onClose app:///../node_modules/engine.io-client/lib/socket.js:725
    setTransport app:///../node_modules/engine.io-client/lib/socket.js:284
    emit app:///../node_modules/engine.io-client/node_modules/component-emitter/index.js:133
    onClose app:///../node_modules/engine.io-client/lib/transport.js:160
    onClose app:///../node_modules/engine.io-client/lib/transports/websocket.js:233
    onclose app:///../node_modules/engine.io-client/lib/transports/websocket.js:153
    getSupportVariation app:///./state/selectors/get-inline-help-support-variation.js:37
    Redux 2
    InlineHelpContactViewLoaded app:///./blocks/inline-help/inline-help-contact-view.jsx:30
    React 9
    unstable_runWithPriority app:///../node_modules/scheduler/cjs/scheduler.development.js:468
    React 4
    Redux 4
    e http://calypso.localhost:3000/plans/klimeryk.wordpress.com?flags=force-sympathy:2494
    dispatch app:///./state/action-log/index.js:98
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/audio/middleware.js:57
    libraryMiddleware app:///./state/lib/middleware.js:81
    analyticsMiddleware app:///./state/analytics/middleware.js:104
    enhancer Redux
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/happychat/middleware-calypso.js:221
    socketMiddleware app:///./state/happychat/middleware.js:69
    middleware app:///./state/data-layer/wpcom-api-middleware.js:73
    Redux 2
    ticketSupportConfigurationRequest app:///./state/help/ticket/actions.js:31
    (Async: promise callback)
    ticketSupportConfigurationRequest app:///./state/help/ticket/actions.js:30
    Redux 2
    request app:///./components/data/query-ticket-support-configuration/index.jsx:16
    createThunkMiddleware Redux
    QueryTicketSupportConfiguration app:///./components/data/query-ticket-support-configuration/index.jsx:23
    React 5
    unstable_runWithPriority app:///../node_modules/scheduler/cjs/scheduler.development.js:468
    React 3
    workLoop app:///../node_modules/scheduler/cjs/scheduler.development.js:417
    flushWork app:///../node_modules/scheduler/cjs/scheduler.development.js:390
    performWorkUntilDeadline app:///../node_modules/scheduler/cjs/scheduler.development.js:157
    (Async: EventHandlerNonNull)
    <anonymous> app:///../node_modules/scheduler/cjs/scheduler.development.js:180
    <anonymous> app:///../node_modules/scheduler/cjs/scheduler.development.js:645
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:888
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///../node_modules/scheduler/index.js:4
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:896
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    React 2
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:864
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> React
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:872
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///./boot/common.js:12
    js http://calypso.localhost:3000/calypso/evergreen/entry-main.js:78
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///./boot/app.js:3
    js http://calypso.localhost:3000/calypso/evergreen/entry-main.js:70
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    __webpack_exec__ http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1396
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1397
    O http://calypso.localhost:3000/calypso/evergreen/runtime.js:80
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1398
    webpackJsonpCallback http://calypso.localhost:3000/calypso/evergreen/runtime.js:486
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:9
```

```
Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.
dispatch@app:///../node_modules/redux/es/redux.js:669:28
request/<@app:///./components/data/query-ticket-support-configuration/index.jsx:16:13
section
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
div
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
InlineHelpPopover
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
div
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
createThunkMiddleware/</</<@app:///../node_modules/redux-thunk/es/index.js:12:18
QueryTicketSupportConfiguration/<@app:///./components/data/query-ticket-support-configuration/index.jsx:23:13
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14
callCallback@app:///../node_modules/react-dom/cjs/react-dom.development.js:3945:14 [react_devtools_backend.js:4061:25](moz-extension://3d369228-a8c1-a84d-a5f9-9fcd2848a917/build/react_devtools_backend.js)
    overrideMethod moz-extension://3d369228-a8c1-a84d-a5f9-9fcd2848a917/build/react_devtools_backend.js:4061
    React 6
    Notifications app:///./notifications/index.jsx:112
    getSupportVariation app:///./state/selectors/get-inline-help-support-variation.js:37
    Redux 2
    InlineHelpContactViewLoaded app:///./blocks/inline-help/inline-help-contact-view.jsx:30
    React 9
    unstable_runWithPriority app:///../node_modules/scheduler/cjs/scheduler.development.js:468
    React 4
    Redux 4
    e http://calypso.localhost:3000/plans/klimeryk.wordpress.com?flags=force-sympathy:2494
    dispatch app:///./state/action-log/index.js:98
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/audio/middleware.js:57
    libraryMiddleware app:///./state/lib/middleware.js:81
    analyticsMiddleware app:///./state/analytics/middleware.js:104
    enhancer Redux
    __WEBPACK_DEFAULT_EXPORT__ app:///./state/happychat/middleware-calypso.js:221
    socketMiddleware app:///./state/happychat/middleware.js:69
    middleware app:///./state/data-layer/wpcom-api-middleware.js:73
    Redux 2
    ticketSupportConfigurationRequest app:///./state/help/ticket/actions.js:31
    (Async: promise callback)
    ticketSupportConfigurationRequest app:///./state/help/ticket/actions.js:30
    Redux 2
    request app:///./components/data/query-ticket-support-configuration/index.jsx:16
    createThunkMiddleware Redux
    QueryTicketSupportConfiguration app:///./components/data/query-ticket-support-configuration/index.jsx:23
    React 5
    unstable_runWithPriority app:///../node_modules/scheduler/cjs/scheduler.development.js:468
    React 3
    workLoop app:///../node_modules/scheduler/cjs/scheduler.development.js:417
    flushWork app:///../node_modules/scheduler/cjs/scheduler.development.js:390
    performWorkUntilDeadline app:///../node_modules/scheduler/cjs/scheduler.development.js:157
    (Async: EventHandlerNonNull)
    <anonymous> app:///../node_modules/scheduler/cjs/scheduler.development.js:180
    <anonymous> app:///../node_modules/scheduler/cjs/scheduler.development.js:645
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:888
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///../node_modules/scheduler/index.js:4
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:896
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    React 2
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:864
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> React
    js http://calypso.localhost:3000/calypso/evergreen/vendors-node_modules_tannin_sprintf_index_js-node_modules_wordpress_compose_build-module_util-df294e.js:872
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///./boot/common.js:12
    js http://calypso.localhost:3000/calypso/evergreen/entry-main.js:78
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    <anonymous> app:///./boot/app.js:3
    js http://calypso.localhost:3000/calypso/evergreen/entry-main.js:70
    __webpack_require__ http://calypso.localhost:3000/calypso/evergreen/runtime.js:31
    __webpack_exec__ http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1396
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1397
    O http://calypso.localhost:3000/calypso/evergreen/runtime.js:80
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:1398
    webpackJsonpCallback http://calypso.localhost:3000/calypso/evergreen/runtime.js:486
    <anonymous> http://calypso.localhost:3000/calypso/evergreen/entry-main.js:9
```

I've tried following the debug instructions there, but I'm honestly getting lost in all the layers here 😅 What's more confusing, is that this was working fine for my a8c user and then it stopped. I tried loading Calypso with fresh state (`?flags=force-sympathy`), but the issue persists. Interestingly, for my non-a8c user, I cannot reproduce (or trigger the error). And then while writing this comment, the issue on the a8c account disappeared 😅 

Steps to reproduce:
 - Go to My Sites -> Plan. Make sure your site has a paid plan.
 - Click the ? fab in lower right corner.
 - Click Contact Us.

The inline form popup would then be stuck in loading mode and the error would show up (I think after the `support-history` API call returns).

The only thing that I played with is the `persist: false` parameter that I changed to `true` (to see what it does ;)). But reverting it did not help, at least initially. And even with `true`, I could not reproduce on the other user.

So, all in all, quite confused and afraid to deploy this to production. I wonder if it's a rare race condition or something. Would love your expert opinion here 🙇‍♂️

tyxla commented:
@klimeryk thanks for all the details here! I tried reproducing with an a8c account and a non-a8c account, and was unable to reproduce it even once. I actually tried other ways to get to happychat (and contact form with chat disabled), and couldn't repro it even a single time. 

It's quite possible this was a one-off scenario caused by stale cache, but we can't tell for sure without a consistent way to reproduce it. Let me know if you run into it again!

That being said, I think this is good to go 😉
